packages/jellyfish-transaction/src/script/dftx/dftx_icxorderbook.ts

```diff
@@ -0,0 +1,50 @@
+import { BufferComposer, ComposableBuffer } from '../../buffer/buffer_composer'
+import BigNumber from 'bignumber.js'
+import { Script } from '../../tx'
+import { CScript } from '../../tx_composer'
+
+// Disabling no-return-assign makes the code cleaner with the setter and getter */
+/* eslint-disable no-return-assign */
+
+export enum ICXOrderType {
+  /** type for DFI/BTC orders */
+  INTERNAL = 1,
+  /** type for BTC/DFI orders */
+  EXTERNAL = 2
+}
+
+/**
+ * ICX CreateOrder DeFi Transaction
+ */
+
+export interface ICXCreateOrder {
+  orderType: number // -------------| 1 byte unsigned
+  tokenId: number // ---------------| VarUInt{1-9 bytes}
+  ownerAddress: Script // ----------| n = VarUInt{1-9 bytes}, + n bytes
+  receivePubkey?: string // --------| 1 byte for len + len bytes
```

Comment:
```suggestion
  receivePubkey?: string // --------| 1-n bytes, first byte value = length = n - 1
```
As many of us actually confused. It is actually optional after deserialized, it has at least 1 byte in buffer.

Reply:
receivePubkey?: string // --------| 1 byte for len (not optional) + len bytes ?

Reply:
I am not very picky on this, just trying stick to how other documented, like the `VarUInt{1-9 bytes}` to show it is actually not "optional" as 0x00 will be in the buffer regardless.